src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java

```diff
@@ -108,24 +114,49 @@ private OneTimePassword getOneTimePassword(OneTimePasswordDto oneTimePasswordDto
 
     @Override
     public void check(OneTimePasswordCheckDto oneTimePasswordCheckDto) {
-
-        OneTimePassword otp = oneTimePasswordRepository.findById(oneTimePasswordCheckDto.getId())
+        // validate
+        OneTimePassword otp = oneTimePasswordRepository
+            .findById(oneTimePasswordCheckDto.getId())
             .orElseThrow(OtpInvalidPasswordException::new);
-
-        if (checkOtpState(otp)
-            || checkOtpDate(otp)
-            || checkOtpRetries(otp)
-            || checkOtpPasswd(otp, oneTimePasswordCheckDto)) {
-
+        try {
+            if (checkOtpState(otp)) {
```

Comment:
please throw each exception inside each of these `check` methods

example: 
```java 
    private void checkOtpState(OneTimePassword otp) {
        if (otp.getStateKey() != StateKey.ACTIVE) {
              throw new IllegalOtpStateException();
        } 
    }
```

it'll result in much cleaner code here in main `check` method

Reply:
fixed